ext/hash/php_hash.h

```diff
@@ -131,6 +148,15 @@ extern PHP_HASH_API zend_class_entry *php_hashcontext_ce;
 PHP_HASH_API const php_hash_ops *php_hash_fetch_ops(zend_string *algo);
 PHP_HASH_API void php_hash_register_algo(const char *algo, const php_hash_ops *ops);
 PHP_HASH_API int php_hash_copy(const void *ops, void *orig_context, void *dest_context);
+PHP_HASH_API int php_hash_serialize(const php_hashcontext_object *context, zend_long *magic, zval *zv);
+PHP_HASH_API int php_hash_unserialize(php_hashcontext_object *context, zend_long magic, const zval *zv);
+
+static inline void *php_hash_alloc_context(const php_hash_ops *ops) {
```

Comment:
I'm okay with keeping this helper function in place to have a common path for all context allocations, but FYI `ecalloc()` does exactly this.  Can replace the body of this helper with the following:

```
return ecalloc(1, ops->context_size);
```

Reply:
Did this in fd6ffc8, which also fixes alignments.